### PR TITLE
[FIX] Update diff highlighting

### LINKED ISSUE #325: Contrast problem in diffeditor
Currently used `diffEditor.*LineBackground` and `diffEditor.*TextBackground` colors are barely distinguishable in both GitHub Dark and GitHub Light themes:
image
image
For example, here are screenshots of vscode builtin themes:
image
image
image
github.com diffeditor uses more contrasting colors as well:
image

### PR DESCRIPTION
**Pull request by Mekomancer**
Closes #325

### CODE REVIEW

File: src/theme.js [hunk 1]
```diff
@@ -208,10 +208,10 @@ function getTheme({ theme, name }) {
...
       "editorGutter.deletedBackground" : color.danger.muted,
 
-      "diffEditor.insertedLineBackground": lightDark(alpha(scale.green[1], 0.3), alpha(scale.green[5], 0.2)),
-      "diffEditor.insertedTextBackground": lightDark(alpha(scale.green[2], 0.4), alpha(scale.green[5], 0.3)),
-      "diffEditor.removedLineBackground" : lightDark(alpha(scale.red[1], 0.3), alpha(scale.red[5], 0.2)),
-      "diffEditor.removedTextBackground" : lightDark(alpha(scale.red[2], 0.4), alpha(scale.red[5], 0.3)),
+      "diffEditor.insertedLineBackground": color.diffBlob.addition.lineBg,
+      "diffEditor.insertedTextBackground": color.diffBlob.addition.textBg,
+      "diffEditor.removedLineBackground" : color.diffBlob.deletion.lineBg,
+      "diffEditor.removedTextBackground" : color.diffBlob.deletion.textBg,
```
Comment (Mekomancer) [inline src/theme.js:214]: Updated diff highlighting for better contrast
See issue #325 and pr #328

**Comment by lukasoppermann** [comment]
Hey @Mekomancer I think this sadly does not work because there are a lot of possible layering problems where this color is layered on top of other things that need to be visible through the color.

**Comment by simurai** [comment]
> where this color is layered on top of other things

👍 Yes, most of the backgrounds need to be semi-transparent so that underlaying layers remain visible. Things like selections or bracket/word matcher. From the [VS Code docs](https://code.visualstudio.com/api/references/theme-color#diff-editor-colors):

> The color must not be opaque so as not to hide underlying decorations.

It's a shame we can used the colors from Primitives, but the VS Code theme is a bit an edge case in that regard.

**Comment by simurai** [comment]
Took another look. The current diff should already have more contrast than the screenshots in https://github.com/primer/github-vscode-theme/issues/325, but with https://github.com/primer/github-vscode-theme/pull/328/commits/1be76bb40016386dcb4e0b3fbcb6d94467203afa I increased the contrast for diff colors a bit more, but still kept it semi-transparent. Here the changes:

github.com | Before | After
--- | --- | ---
dotcom | before | after
dotcom-d | before-d | after-d

**Pull request merged**